src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems

```diff
@@ -1446,7 +1446,7 @@
   <ItemGroup Condition="'$(FeatureCominterop)' == 'true'">
     <Compile Include="$(MSBuildThisFileDirectory)System\Runtime\InteropServices\WindowsRuntime\EventRegistrationToken.cs" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetsWindows)' == 'true' or '$(FeaturePal)'=='true'">
+  <ItemGroup Condition="'$(TargetsWindows)' == 'true' or '$(FeatureCoreCLR)'=='true'">
     <Compile Include="$(MSBuildThisFileDirectory)Microsoft\Win32\SafeHandles\SafeWaitHandle.Windows.cs" />
```

Comment:
This is a bit strange that non-Windows coreclr needs .Windows files

Reply:
Looks like this was originally introduced in dotnet/coreclr#13492 to for 3 windows files to share exceptions.

```diff
+  <ItemGroup Condition="$(TargetsWindows) or '$(FeaturePal)'=='true'">
+    <Compile Include="$(MSBuildThisFileDirectory)Interop\Windows\Interop.Errors.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)Interop\Windows\Kernel32\Interop.FormatMessage.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)System\IO\Win32Marshal.cs" />
+  </ItemGroup>
```

Reply:
@stephentoub @Anipik Grew @danmosemsft's original set with these commits.

dotnet/coreclr#22540
dotnet/coreclr#22106
dotnet/coreclr#18662
dotnet/coreclr#18612
dotnet/coreclr#18597
dotnet/coreclr#18320

Reply:
Opened issue #31721 to resolve this.

Reply:
Could you please add a comment that says "CoreCLR uses PAL layer that emulates Windows API on Unix. This is bridge for that PAL layer.". And link to the issue you have opened.